src/SwedbankPay.Sdk.Infrastructure/PaymentInstruments/Invoice/InvoicePaymentOperations.cs

```diff
@@ -1,71 +1,69 @@
-                        ViewAuthorization = httpOperation;
-                        break;
-
-                    case PaymentResourceOperations.CreateCapture:
-                        Capture = async payload =>
-                            await client.SendAsJsonAsync<CaptureResponse>(httpOperation.Method, httpOperation.Href, payload);
-                        break;
-
-                    case PaymentResourceOperations.CreateCancellation:
-                        Cancel = async payload =>
-                            await client.SendAsJsonAsync<CancellationResponse>(httpOperation.Method, httpOperation.Href, payload);
-                        break;
-
-                    case PaymentResourceOperations.CreateReversal:
-                        Reversal = async payload =>
-                            await client.SendAsJsonAsync<ReversalResponse>(httpOperation.Method, httpOperation.Href, payload);
-                        break;
-
-                    case PaymentResourceOperations.CreateApprovedLegalAddress:
-                        ApprovedLegalAddress = async payload =>
-                            await client.SendAsJsonAsync<ApprovedLegalAddressResponse>(httpOperation.Method, httpOperation.Href, payload);
-                        break;
-
-                    case PaymentResourceOperations.DirectAuthorization:
-                        DirectAuthorization = async payload =>
-                            await client.SendAsJsonAsync<InvoicePaymentAuthorizationResponse>(httpOperation.Method, httpOperation.Href, payload);
-                        break;
-                }
-                this.Add(httpOperation.Rel, httpOperation);
-            }
-        }
-
-        public Func<InvoiceApprovedLegalAddressRequest, Task<ApprovedLegalAddressResponse>> ApprovedLegalAddress{ get; }
-        public Func<PaymentAbortRequest, Task<InvoicePaymentResponse>> Abort { get; }
-        public Func<InvoicePaymentCancelRequest, Task<CancellationResponse>> Cancel { get; }
-        public Func<InvoicePaymentCaptureRequest, Task<CaptureResponse>> Capture { get; }
-        public Func<InvoicePaymentAuthorizationRequest, Task<InvoicePaymentAuthorizationResponse>> DirectAuthorization { get; }
-        public Func<InvoicePaymentReversalRequest, Task<ReversalResponse>> Reversal { get; }
-        public HttpOperation RedirectAuthorization { get; }
-        public HttpOperation ViewAuthorization { get; }
-
-    }
+﻿using System;
+using SwedbankPay.Sdk.Extensions;
+using System.Threading.Tasks;
+using System.Net.Http;
+using SwedbankPay.Sdk.Common;
+
+namespace SwedbankPay.Sdk.PaymentInstruments.Invoice
+{
+    public class InvoicePaymentOperations : OperationsBase, IInvoicePaymentOperations
+    {
+        public InvoicePaymentOperations(IOperationList operations, HttpClient client)
+        {
+            foreach (var httpOperation in operations)
+            {
+                switch (httpOperation.Rel.Value)

```

Comment:
Codacy found an issue: [Add a 'default' clause to this 'switch' statement.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-dotnet/pullRequest?prid=6290774)